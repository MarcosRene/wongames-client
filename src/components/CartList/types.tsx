import { GameItemProps } from 'components/GameItem/types'

export type CartListProps = Partial<{
  items: GameItemProps[]
  total: string
  hasButton: boolean
}>
