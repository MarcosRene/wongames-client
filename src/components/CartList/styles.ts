import styled, { css } from 'styled-components'
import media from 'styled-media-query'
import { tint } from 'polished'

export const Wrapper = styled.main`
  ${({ theme }) => css`
    background: ${theme.colors.white};

    display: flex;
    flex-direction: column;
    align-self: start;
  `}
`

export const Footer = styled.div`
  ${({ theme }) => css`
    padding: 2rem;
    font-weight: ${theme.font.bold};
    font-size: ${theme.font.sizes.small};
    background: ${tint(0.2, theme.colors.lightGray)};
    color: ${theme.colors.black};

    display: flex;
    justify-content: space-between;
    align-items: center;

    ${media.greaterThan('medium')`
      padding: ${theme.spacings.small};
      font-size: ${theme.font.sizes.medium};
    `};
  `}
`

export const Total = styled.span`
  ${({ theme }) => css`
    color: ${theme.colors.primary};
  `}
`
