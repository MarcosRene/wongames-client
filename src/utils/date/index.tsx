export default function formatedDate(releaseDate: string) {
  return new Intl.DateTimeFormat('en-US', {
    day: 'numeric',
    month: 'short',
    year: 'numeric'
  }).format(new Date(releaseDate))
}
