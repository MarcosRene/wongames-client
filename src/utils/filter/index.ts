import { ParsedUrlQueryInput } from 'querystring'
import { ItemProps } from 'components/ExploreSidebar/types'

type ParseArgs = {
  queryString: ParsedUrlQueryInput
  filterItems: Pick<ItemProps, 'type' | 'name'>[]
}

export function parseQueryStringToFilters({
  queryString,
  filterItems
}: ParseArgs) {
  const obj: Record<string, string> = {}

  for (const key of Object.keys(queryString).filter(
    (item) => item !== 'sort'
  )) {
    const item = filterItems.find((filterItem) => filterItem.name === key)
    const isCheckbox = item?.type === 'checkbox'

    obj[key] = (
      !isCheckbox
        ? queryString[key]
        : {
            name: queryString[key]
          }
    ) as string
  }

  return obj
}

export function parseQueryStringToFilter({
  queryString,
  filterItems
}: ParseArgs) {
  const obj: Record<string, string[]> = {}

  for (const key of Object.keys(queryString)) {
    const type = filterItems.find((filterItem) => filterItem.name === key)?.type
    const isCheckbox = type === 'checkbox'
    const isArray = Array.isArray(queryString[key])

    obj[key] =
      !isArray && isCheckbox
        ? [queryString[key] as string]
        : (queryString[key] as string[])
  }

  return obj
}
