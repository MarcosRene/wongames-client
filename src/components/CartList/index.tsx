import GameItem from 'components/GameItem'

import { CartListProps } from './types'

import * as S from './styles'
import Button from 'components/Button'
import Link from 'next/link'

const CartList = ({ items, total, hasButton = false }: CartListProps) => {
  console.log({ hasButton })
  return (
    <S.Wrapper>
      {items.map((item) => (
        <GameItem key={item.title} {...item} />
      ))}

      <S.Footer>
        {!hasButton && <span>Total: </span>}
        <S.Total>{total}</S.Total>
        {hasButton && (
          <Link href="/cart">
            <Button as="a">Buy it now</Button>
          </Link>
        )}
      </S.Footer>
    </S.Wrapper>
  )
}

export default CartList
