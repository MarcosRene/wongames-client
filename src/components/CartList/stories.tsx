import { Meta, StoryObj } from '@storybook/react'
import styled from 'styled-components'

import CartList from '.'

import cartListMock from './mock'

export default {
  title: 'CartList',
  component: CartList,
  argTypes: {
    items: {
      type: ''
    }
  },
  args: {
    items: cartListMock,
    total: 'R$ 330,00'
  }
} as Meta<typeof CartList> | unknown

const Container = styled.div`
  max-width: 80rem;
`

export const Default: StoryObj<typeof CartList> = {
  render: (args) => (
    <Container>
      <CartList {...args} />
    </Container>
  )
}

export const WithButton: StoryObj<typeof CartList> = {
  render: (args) => (
    <Container>
      <CartList {...args} hasButton />
    </Container>
  )
}

export const Empty: StoryObj<typeof CartList> = {
  render: () => (
    <Container>
      <CartList />
    </Container>
  )
}
