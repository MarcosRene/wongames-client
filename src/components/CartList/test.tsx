import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests'
import theme from 'styles/theme'

import CartList from '.'

import cartListMock from './mock'

describe('<CartList />', () => {
  it('should render the cart list ', () => {
    const { container } = renderWithTheme(
      <CartList items={cartListMock} total="R$ 330,00" />
    )

    expect(screen.getAllByRole('heading')).toHaveLength(2)
    expect(screen.getByText(/R\$ 330,00/i)).toBeInTheDocument()
    expect(screen.getByText(/R\$ 330,00/i)).toHaveStyle({
      color: theme.colors.primary
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the button', () => {
    renderWithTheme(
      <CartList items={cartListMock} total="R$ 330,00" hasButton />
    )

    expect(screen.getByText(/buy it now/i)).toBeInTheDocument()
  })
})
