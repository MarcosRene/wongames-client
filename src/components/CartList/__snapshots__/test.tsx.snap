// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartList /> should render the cart list  1`] = `
.c1 {
  padding: 2.4rem;
  border-bottom: 0.1rem solid #EAEAEA;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  margin-right: 1.2rem;
  width: 9.6rem;
  height: 5.6rem;
  position: relative;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c3 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c5 {
  margin-bottom: 0.8rem;
  font-size: 1.4rem;
  font-weight: 600;
  line-height: 1.4rem;
  color: #030517;
}

.c6 {
  padding: 0.2rem 0.8rem;
  font-size: 1.4rem;
  font-weight: 600;
  background-color: #3CD3C1;
  color: #FAFAFA;
  border-radius: 0.4rem;
}

.c0 {
  background: #FAFAFA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
}

.c7 {
  padding: 2rem;
  font-weight: 600;
  font-size: 1.4rem;
  background: #eee;
  color: #030517;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 {
  color: #F231A5;
}

@media (min-width:768px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c3 {
    width: 15rem;
    height: 7rem;
    margin-right: 1.6rem;
  }
}

@media (min-width:768px) {
  .c5 {
    font-size: 2rem;
    line-height: 2rem;
  }
}

@media (min-width:768px) {
  .c7 {
    padding: 2.4rem;
    font-size: 1.6rem;
  }
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Red Dead Redemption 2"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="c4"
      >
        <h3
          class="c5"
        >
          Red Dead Redemption 2
        </h3>
        <p
          class="c6"
        >
          R$ 215,00
        </p>
      </div>
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="Borderlands 3"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="c4"
      >
        <h3
          class="c5"
        >
          Borderlands 3
        </h3>
        <p
          class="c6"
        >
          R$ 215,00
        </p>
      </div>
    </div>
  </div>
  <div
    class="c7"
  >
    Total 
    <span
      class="c8"
    >
      R$ 330,00
    </span>
  </div>
</main>
`;
